File: 43/다음수 구하기/김진희.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;

public class boj_2697 {
    public static void main(String[] args) throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());

        for (int i = 0; i < n; i++) {
            String input = br.readLine();
            System.out.println(solution(input));
        }
    }

    public static String solution(String a) {
        int[] aArray = Arrays.stream(a.split("")).mapToInt(Integer::parseInt).toArray();
        StringBuilder answer = new StringBuilder();
        int start = aArray.length - 1;
        int end = -1;
        int x = -1;
        int y = -1;

        while (start - 1 > end) {
            for (int j = start - 1; j > end; j--) {
                if (aArray[start] > aArray[j]) {
                    if (x > 0 && end == j && aArray[start] > aArray[x]) {
                        y = j;
                        end = j;
                        break;
                    }
                    x = start;
                    y = j;
                    end = j;
                    break;
                }
            }
            start--;
        }

        if (x == -1) {
            return "BIGGEST";
        }

        int temp = aArray[x];
        aArray[x] = aArray[y];
        aArray[y] = temp;

        int[] bArray = Arrays.copyOfRange(aArray, 0, y + 1);
        int[] sortedArray = Arrays.copyOfRange(aArray, y + 1, aArray.length);
        Arrays.sort(sortedArray);

        int[] mergedArray = new int[bArray.length + sortedArray.length];
        System.arraycopy(bArray, 0, mergedArray, 0, bArray.length);
        System.arraycopy(sortedArray, 0, mergedArray, bArray.length, sortedArray.length);

        for (int value : mergedArray) {
            answer.append(value);
        }

        return answer.toString();
    }
}

File: 47/다오와 트리플 멕스 게임/김진희.py
# 다오와 트리플 멕스 게임
def blank():
    start = 0
    end = n - 1
    while start < end:
        middle = (start + end) // 2
        if a[middle] == middle:
            start = middle
            if end - start == 1:
                return start + 3
        else:
            if a[middle - 1] == middle - 1:
                return middle + 2
            else:
                end = middle


N = int(input())
a = list(set(map(int, input().split())))
n = len(a)
if n == 1:
    print(0) if a[0] else print(1)
else:
    a.sort()
    # 배열에 0이 없으면 0
    if a[0]:
        print(0)
    # 배열에 0이 있을 때
    else:
        if a[-1] == n - 1:
            print(a[-1] + 3)
        else:
            print(blank())
